Add leading silence detection and removal logic (#17648)
nvaccess/nvda: user_docs/en/userGuide.md

In particular moving quickly through messages in Gmail with Chrome can cause NVDA to speak outdated information.
This functionality is enabled by default as of NVDA 2021.1.

> ##### Trim leading silence in speech audio {#TrimLeadingSilence}

Reviewer: Just in case we elect to trim silence in other situations for some reason
```suggestion
##### Trim leading silence in speech audio {#TrimLeadingSilenceSpeech}
```

In particular moving quickly through messages in Gmail with Chrome can cause NVDA to speak outdated information.
This functionality is enabled by default as of NVDA 2021.1.

> ##### Trim leading silence in speech audio {#AdvancedSettingsTrimLeadingSilence}

Reviewer: We try to avoid encoding GUI structure in anchors in case the GUI gets reorganised
```suggestion
##### Trim leading silence in speech audio {#TrimLeadingSilenceSpeech}
```
(You'll also need to update https://github.com/gexgd0419/nvda/blob/c2f98f563d4fd775767f76b0f23e2842c968a48d/source/gui/settingsDialogs.py#L3758)

> This option enables NVDA to trim the silence at the beginning of the speech audio, making the voices respond faster.
> This option is enabled by default, and should only affect the silence at the beginning. But if you find that some necessary silence periods are also missing (e.g. pause between two sentences) when using a speech synthesizer add-on, you may try turning off this feature.

Reviewer: One sentence per line, plus some clarity improvements.
```suggestion
When enabled, NVDA will remove silence from the start of speech audio, which may improve the responsiveness of some speech synthesizers.
This option is enabled by default, and should only affect the silence at the beginning of speech.
If you find that some necessary silence periods are also missing (e.g. pause between two sentences) when using a speech synthesizer add-on, you may turn this feature off entirely to resolve the issue.
```
